.cursor/rules/react.mdc

```diff
@@ -125,4 +125,22 @@ globs: **/*.jsx,**/*.tsx
 - Keep components focused on a single responsibility
 - Document complex logic with clear comments
 - Follow the project's folder structure and naming conventions
-- Prefer controlled components over uncontrolled ones 
\ No newline at end of file
+- Prefer controlled components over uncontrolled ones
+
+## Readability & Formatting
+
+- Always run `make fmt-all` (Prettier/Biome/Ruff) before committing.
+- Use spaces around binary operators and after commas: `const xs = pts.map(p => p.data.x);` not `pts.map(p=>p.data.x)`.
+- Limit line length to 100 chars; break long expressions.
+
+## Declarative Rendering vs Imperative Canvas
+
+- Prefer **declarative SVG** (or a React-first charting lib such as `@visx/xy-chart`, `recharts`, `nivo`) for 2-D data‐viz inside React.  
```

Comment:
That's very specific and should be used for exact tasks. Adding it here will force agent to add new libds instead of reusing Konva.js or d3.js that we already have in our codebase